pubsub/ws.go

```diff
@@ -142,9 +151,18 @@ func (w *WsPubSub) reader(c *connection) {
 			}
 			c.channels = append(c.channels, payload.Channel)
 		case "pub":
-			w.hub.Broadcast <- Parcel{
-				Channel: payload.Channel,
-				Data:    []byte(*payload.Data),
+			if payload.Data == nil {
+				log.Debugf("Got nil pub data.")
+				c.ws.WriteMessage(
+					websocket.TextMessage,
+					[]byte("Error: missing data to publish. Closing Connection"),
+				)
+				c.ws.WriteMessage(websocket.CloseMessage, nil)
+			} else {
```

Comment:
I think the convention in golang is for error scenario to return early so the normal path has a lower indentation level.